src/Application/Model/Ticket.php

```diff
@@ -851,6 +851,36 @@ public function resetSource(Comment $comment = null) {
 			
 			return true;
 		}
+
+		public function reset($handle_id = null, $ticket_types = []) {
+			$tickets = Database::$Instance->query(
+				'SELECT * FROM ticket_reset(?,?)',
+				[
+					$this['id'],
+					'{' . implode(',', $ticket_types) . '}'
```

Comment:
@jjeising Does this work properly? Enough escaping etc. so this won't lead to SQL injections?

Reply:
Sorry, I've missed this one. This uses [PDO::prepare](http://php.net/manual/en/pdo.prepare.php), so if PostgreSQL supports arrays in prepared parameters (as it was hopefully tested by @j-licht ) this should be fine. Would be interesting to test, what happens if a user supplied ticket_type contains braces.

Reply:
When I add a brace or comma, I got:
`PHP Fatal error:  Uncaught Exception: ERROR:  malformed array literal`

Reply:
Ok. There are probably some other cases an user could trigger a SQL Exception (numeric values or similar), but in this case I would feel better if all ticket types fed into this query exist.

Reply:
@pegro are you fine with this SQL building?